cloud/dimensiondata/dimensiondata_vlan.py

```diff
@@ -0,0 +1,411 @@
+            type: string
+            sample: NA3
+        status:
+            description: VLAN status.
+            type: string
+            sample: NORMAL
+'''
+
+
+def vlan_obj_to_dict(vlan):
+    vlan_d = dict(id=vlan.id,
+                  name=vlan.name,
+                  description=vlan.description,
+                  location=vlan.location.id,
+                  private_ipv4_range_address=vlan.private_ipv4_range_address,
+                  private_ipv4_range_size=vlan.private_ipv4_range_size,
+                  status=vlan.status,
+                  )
+    return vlan_d
+
+
+def wait_for_vlan_state(module, driver, vlan_id, state_to_wait_for):
+    try:
+        return driver.connection.wait_for_state(
+            state_to_wait_for, driver.ex_get_vlan,
+            module.params['wait_poll_interval'],
+            module.params['wait_time'], vlan_id
+        )
+    except DimensionDataAPIException as e:
+        module.fail_json(msg='VLAN did not reach % state in time: %s'
+                         % (state, e.msg))
+
+
+def create_vlan(module, driver, location, network_domain, name, description,
+                base_address, prefix_size):
+    try:
+        vlan = driver.ex_create_vlan(network_domain, name, base_address,
+                                     description, prefix_size)
+    except DimensionDataAPIException as e:
+        if e.code == 'NAME_NOT_UNIQUE':
+            vlan = get_vlan(module, driver, location, network_domain, 'False',
+                            name)
+            return {'vlan': vlan, 'changed': False}
+        module.fail_json(msg="Failed to create VLAN: %s" % e)
+    # Wait for it to become ready
+    if module.params['wait']:
+        vlan = wait_for_vlan_state(module, driver, vlan.id, 'NORMAL')
+    return {'vlan': vlan, 'changed': True}
+
+
+def get_vlan(module, driver, location, network_domain, vlan_id, name):
+    if vlan_id is not 'False':
+        try:
+            vlan = driver.ex_get_vlan(vlan_id)
+            return vlan
+        except DimensionDataAPIException as e:
+            if e.code == "NOT_FOUND":
+                return {'id': False}
+            else:
+                module.fail_json(msg="Unexpected error: %s" % e)
+    elif name is not False:
+        try:
+            # Find vlan by name
+            vlans = driver.ex_list_vlans(location, network_domain, name)
+            # VLAN not found
+            if len(vlans) < 1:
+                return {'id': False}
+            return driver.ex_get_vlan(vlans[0].id)
+        except DimensionDataAPIException as e:
+            module.fail_json(msg="Unexpected API error: %s" % e)
+    else:
+        module.fail_json(msg="Unexpected Error: One of vlan_id or name " +
+                             "must be provided.")
+
+
+def delete_vlan(module, driver, location, network_domain, vlan_id, name):
+    vlan = get_vlan(module, driver, location, network_domain,
+                    vlan_id, name)
+    if type(vlan) is dict:
```

Comment:
better to do
``` python
if isinstance(vlan, dict)
```
If this module is restricted to python2.6 and above because of the library, even better to do
``` python
from collections import MutableMapping
[...]
if isinstance(vlan, MutableMapping):
```